lib/MemoryModel/LocMemModel.cpp

```diff
@@ -94,19 +94,19 @@ bool LocSymTableInfo::computeGepOffset(const llvm::User *V, LocationSet& ls) {
             // Handling pointer types
             // These GEP instructions are simply making address computations from the base pointer address
             // e.g. idx1 = (char*) &MyVar + 4,  at this case gep only one offset index (idx)
-            if (const PointerType* pty = dyn_cast<PointerType>(*gi)) {
+            if (const PointerType* pty = dyn_cast<PointerType>(gi.getIndexedType())) {
                 const Type* et = pty->getElementType();
                 Size_t sz = getTypeSizeInBytes(et);
                 ls.offset += idx * sz;
             }
             // Calculate the size of the array element
-            else if(const ArrayType* at = dyn_cast<ArrayType>(*gi)) {
+            else if(const ArrayType* at = dyn_cast<ArrayType>(gi.getIndexedType())) {
                 const Type* et = at->getElementType();
                 Size_t sz = getTypeSizeInBytes(et);
                 ls.offset += idx * sz;
             }
             // Handling struct here
-            else if (const StructType *ST = dyn_cast<StructType>(*gi)) {
```

Comment:
Shall we use `gi.getStructTypeOrNull`? I am not very sure whether these two are equivalent or not. Please keep the original one as you did in `MemModel.cpp`